homeassistant/helpers/config_validation.py

```diff
@@ -1074,6 +1075,56 @@ def validator(config: dict) -> dict:
     return validator
 
 
+def no_yaml_config_schema(domain: str) -> Callable[[dict], dict]:
+    """Return a config schema which logs if attempted to setup from YAML."""
+
+    module = inspect.getmodule(inspect.stack(context=0)[2].frame)
+    if module is not None:
+        module_name = module.__name__
+    else:
+        # If Python is unable to access the sources files, the call stack frame
+        # will be missing information, so let's guard.
+        # https://github.com/home-assistant/core/issues/24982
+        module_name = __name__
+    logger_func = logging.getLogger(module_name).error
+
+    def raise_issue() -> None:
+        # pylint: disable-next=import-outside-toplevel
+        from .issue_registry import IssueSeverity, async_create_issue
+
+        with contextlib.suppress(LookupError):
+            hass = async_get_hass()
+        if not hass:
+            return
```

Comment:
Do we need this check?

Reply:
I prefer to have that check there to not raise if the validator is called from the wrong context

Reply:
Should we add a test then